src/Skia/Avalonia.Skia/DrawingContextImpl.cs

```diff
@@ -53,14 +57,48 @@ public void DrawGeometry(IBrush brush, Pen pen, Geometry geometry)
             }
         }
 
-        struct PaintWrapper : IDisposable
+        private struct PaintDisposable : IDisposable
+        {
+            private Action _action;
+
+            public PaintDisposable(Action action)
+            {
+                _action = action;
+            }
+
+            public void Dispose()
+            {
+                _action?.Invoke();
+            }
+        }
+
+        internal struct PaintWrapper : IDisposable
         {
             //We are saving memory allocations there
             //TODO: add more disposable fields if needed
-
             public readonly SKPaint Paint;
+
             private IDisposable _disposable1;
 
+            private Func<SKPaint, SKPaint, IDisposable> _applyToAction;
+
+            public void SetApplyToAction<T>(Func<SKPaint, T> get,
```

Comment:
That's at least 4 heap allocations (one for captured state, 3 for delegates). And that's in a place where we are using structs and stacks to avoid said allocations.